### ISSUE TITLE
Resolution result from .jpg LR and .png LR are so different.

### ISSUE THREAD

**Issue by JustinhoCHN** [zsdonghao/SRGAN#18]
I've trained the SRGAN model for a week, my training config is : 
- **dataset** : DVI2K + my own dataset, about 4000 training pics
- **epochs** : 2000
- **n_epoch_init** : 300
- **training time** : 1 week

I tested some pics, if I use the DVI2K images, it gives pretty good result:

![from DVI2K data](http://ormr426d5.bkt.clouddn.com/17-8-17/93738983.jpg)

but when I test with my images, the result is not so good:

![left: LR, right: generated](http://ormr426d5.bkt.clouddn.com/17-8-17/20381845.jpg)

![left: LR, right: generated](http://ormr426d5.bkt.clouddn.com/17-8-17/62830641.jpg)

I start thinking if that's because the image format is jpg?

As we all know, jpg format picture is **lossy compression** picture, but png format is **lossless compression** picture. jpg throwing away many information for saving disk space, while png kept all information, so I did the fllowing experiment:

I chose a jpg format image as high resolution image, and then compressed them to jpg and png format low-resolution images, and put these two LR image into trained SRGAN model, let's see what I got:

left: jpg LR, right: generated: 
![left: jpg LR, right: generated](http://ormr426d5.bkt.clouddn.com/17-8-18/86960685.jpg)

left: png LR, right: generated:
![left: png LR, right: generated](http://ormr426d5.bkt.clouddn.com/17-8-18/47910197.jpg)

It's obvious that the png format result is better than the jpg format, we can put them in one picture for comparing: 

![left: jpg, right : png](http://ormr426d5.bkt.clouddn.com/17-8-18/27663000.jpg)

**So the question is, are these STOA super resolution papers (including SRGAN) can only work on png format image super resolution? Are there any paper also doing well on jpg images?**

Any ideas will be appreciated, I am working on solving the jpg format images super resolution problem recently.

**Comment by parisburn** [comment]
Did you try to convert a jpg to a png and then run a test? I am not sure why the LR jpg gives so poor quality.

**Comment by JustinhoCHN** [comment]
@parisburn of course I did, I use a ** jpg format image as high resolution image**, and then convert to 1)jpg LR image, 2)png LR image, and put them to the same model.

**Comment by zsdonghao** [comment]
hi, do you have human images in your training set?  because DVI2K doesn't have any human images.

**Comment by JustinhoCHN** [comment]
@zsdonghao yes, there are a lot of human in my trainging set. I added my own dataset which consists of a lot of human images to DVI2K dataset, but the quality is different, my HR dataset is 300kb/image, DVI2K HR is about 4mb/image.

**Comment by zsdonghao** [comment]
I see, I think your training images are too small, I suggest you to use image more than 1000+ pixels.

**Comment by suke27** [comment]
hi, do you have any new result for this problem, I also found there are difference between jpg and png

**Comment by JustinhoCHN** [comment]
@suke27 there's no way to solve this problem for now, because jpg is loss compression, it throws away a lot of image information, while png preserved all information.

**Comment by splinter22** [comment]
Will some preprocess be helpful? For example, make the LR(input) of dataset more jpeg artifacts for training，and the model will learn to reduce these blocking artifacts?

**Comment by suke27** [comment]
anyone has resolved this issue, I also try many method, not work

**Comment by ontheway16** [comment]
@suke27 The only option I can think of is preprocessing the jpegs to clean jpeg artifacts. I remember some software exist for this job. I am pretty sure the block artifacts are primary target to remove for such software.

**Comment by DTennant** [comment]
There is a ICCV2017 paper Deep Generative Adversarial Compression Artifact Removal(https://arxiv.org/abs/1704.02518) dealing with the JPEG compression artifact removal

**Comment by JustinhoCHN** [comment]
@DTennant Thanks a lot! I'll check this out and see what we can do.

**Comment by CasdDesnDR** [comment]
@JustinhoCHN  Hello sir, I am new in super resolution and interested in it. In my opinion, the main function of super resolution is to super resolve the size of image, while maintaining the image quality, such as visual perception. However, in the picture you show,  the LR and generated image have the same size (height and width) . In your shown image, I wonder if you super resolve the LR image  in size through bicubic or other interpolation methods for comparsion. (I know the size of input of the SRGAN  is actually smaller than the output by 2x)

**Comment by JustinhoCHN** [comment]
@CasdDesnDR Yes you are right, in comparison, you have to resize the LR image to the HR size, using bicubic or other methods. Because you can't distinguish the LR and HR image in their original size in human visual perception, and that's the purpose of super-resolution algorithm: if the bicubic method is good, why we still spend our time to find another algorithm?

**Comment by Heermosi** [comment]
@JustinhoCHN Is there any progress made by applying artifact removal?
I've tried the same route as you did, I found the bilateral filter might help reduce some noise in the final result, though they did not help identification.
My Boss now forgive me for not able to SR and identify objects from jpg LR pics.
I thought if that way works, maybe I'll have an alternative to test.

**Comment by JustinhoCHN** [comment]
@Heermosi There're 2 papers I'd like to recommend: [Deep Generative Adversarial Compression Artifact Removal](http://arxiv.org/abs/1704.02518), removing compression artifact using GAN method. And [Learning a Single Convolutional Super-Resolution Network for Multiple Degradations](http://arxiv.org/abs/1712.06116), they proposed that the generalization problem is caused by setting the same downsampling method to build the training dataset, we should consider every possible degradations.

**Comment by Heermosi** [comment]
@JustinhoCHN  Fine, I also suspected current test approach is not applicable in real scene. It's too closely coupled with quality. Which means it was just a local working engine.

**Comment by Heermosi** [comment]
@JustinhoCHN We've tested on raw images captured by Cannon Camera, the pic quality is lower than pics zoomed from HR images. May be the focus is the problem? I guess it was hard to take a pic the same quality as training LR images.

**Comment by Heermosi** [comment]
@JustinhoCHN 
We've found the quality of HR picture is different from quality of LR pics used for training. The zoomed LR pics used for training has a sharper edge for HR reconstruction.
For example, if you take pics from different distances, the origin images would show the same thickness in edge, it's around 2 pixels. while in zoomed pics, they would be only 1 pixel or less.
The optical system seems to have something to do with the sharpness of edges. Even if you thought it might be overcome using optical zooming... there is no effect, smaller object still take 2 pixels for an edge.

It's not only a problem caused by pic formats I think, it has something to do with the optical system.

**Comment by neilthefrobot** [comment]
The issue is definitely .jpg artifacts and I was able to get around this by taking my HR training set, down sampling 4x, then converting it to .jpg with a low quality (high compression) and using that as my LR set. This way the network is seeing jpg artifacts as inputs and a non jpg artifact version as a target and it learns to convert between them. It actually worked very well too.
